changelog_gen.py

```diff
@@ -0,0 +1,189 @@
+import pygit2, requests, re 
+
+# GitHub API stuff
+API_ENDPOINT  = 'https://api.github.com'
+AUTHORISATION_HEADERS = {'Authorization' : ''} # specify auth headers
+
+REPO = '/home/ank/gsoc16/dummy_project' # Specify the repository name here
+USER = 'black-perl'
+PROJECT_REPO = 'dummy_project'
+commit_list = [] # List of commits
+commit_lookup = {} # A hash to lookup whether a commit is considered in the changelog or not
+pull_requests_ids = []
+repo = pygit2.Repository(REPO)
+
+# changelog entries lists
+enhancement_entries = []
+bugfix_entries = []
+merged_pulls_entries = []
+fixed_issues_entries = []
+simple_commits_entries = []
+# a dictionary corresponding to changelog
+changelog_map = {
+    "enhancement" : enhancement_entries,
+    "bugfix" : bugfix_entries,
+    "merged_pull" : merged_pulls_entries,
+    "simple_commit" : simple_commits_entries,
+    "fixed_issue" : fixed_issues_entries
+}
+
+class Entry:
+    '''
+        A changelog entry can be of one of the following category:
+        - enchancement
+        - bugfix
+        - fixed_issues
+        - merged_pulls
+        - simple_commits
+    '''
+    def __init__(self,category):
+        self.link = ''
+        self.text = ''
+        self.author_email = ''
+        self.author_name = ''
+        if category in changelog_map.keys():
+            changelog_map[category].append(self)
+        else:
+            raise Exception('Invalid cateory specified')
+
+    def addText(self,text):
+        self.text = text
+
+    def addLink(self,link):
+        self.link = link
+
+    def addMeta(self,name,email):
+        '''
+            Add meta data incase it is an external 
+        '''
+        self.author_name = name
+        self.author_email = email
+
+
+def get_pull_details(user,repo_name,pull_id):
+    '''
+        Fetches the pull request details for a given pull request id. Two cases basically
+        - Pull request contains commits that closes some issues
+          (these issues may further be categorised as bugs,enhancements if they have label)
+        - Pull request doesn't involve any issue
+    '''
+    request_url = '{0}/repos/{1}/{2}/pulls/{3}/commits'.format(API_ENDPOINT,user,repo_name,pull_id)
+    return requests.get(request_url,headers=AUTHORISATION_HEADERS).json()
+
+def get_issue_details(user,repo_name,issue_id):
+    '''
+        An issue can be one of the following:
+        - Bug 
+        - Enhancement
+        - Others
+    '''
+    request_url = '{0}/repos/{1}/{2}/issues/{3}'.format(API_ENDPOINT,user,repo_name,issue_id)
+    return requests.get(request_url,headers=AUTHORISATION_HEADERS).json()
+
```

Comment:
PRs are updated automatically based on the source branch. I suggest you to use separate branches when you are implementing a feature. Because you send us pull requests from your master branch, your [previous PRs](https://github.com/lbudai/syslog-ng-autorel/pull/1) will be messed up.